index.html

```diff
@@ -49,6 +70,55 @@
 
       var selectedConfigs = new Set();
 
+      function loadVersions(data) {
+      }
+
+      function navigateToVersion(version) {
+        if(!version) {
+          return // if no version present, do not navigate
+        }
+        let params = new URLSearchParams(document.location.search);
+        params.delete("version");
+        params.append("version", version);
+        url = window.location.pathname + `?${params.toString()}`;
+        window.location.href = url;
+      }
+
+      function populateVersionOptions(data) {
+        data = JSON.parse(data)
+        let resp = Array.from(data);
+        let versions = resp.map((res)=>res.sha)
+        const vms = document.getElementById("versionModalSelector");
+        versions.forEach(version=>{
+          const optionElement = document.createElement('option');
+          optionElement.value = optionElement.textContent = version;
+          vms.append(optionElement);
+        })
+
```

Comment:
SHAs are not very user-friendly. Ideally, the user should be given a list of dates to choose from, or even a date picker, with the allowable dates highlighted. The idea is that "this is the snapshot of kconfigs on this date".

We can get the commit dates from each element's `commit.committer.date` attribute. We should keep only the most recent commit for a particular date.

It's OK for the URL to still reference a SHA - that makes it easier to load the kconfigs from that SHA without first looking anything up from the API. But for display, let's only show things in terms of dates.

Reply:
> SHAs are not very user-friendly. Ideally, the user should be given a list of dates to choose from, or even a date picker, with the allowable dates highlighted. The idea is that "this is the snapshot of kconfigs on this date".

I understand your point, and I agree that adding a Date Picker is an excellent idea for selecting the version from the UI. I will implement this change.